sound/soc/sof/topology.c

```diff
@@ -1522,20 +1522,26 @@ static int sof_widget_load_buffer(struct snd_soc_component *scomp, int index,
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct snd_soc_tplg_private *private = &tw->priv;
 	struct sof_ipc_buffer *buffer;
+	size_t ipc_size = sizeof(*buffer) + sizeof(swidget->comp_ext);
 	int ret;
 
 	buffer = kzalloc(sizeof(*buffer), GFP_KERNEL);
 	if (!buffer)
 		return -ENOMEM;
 
 	/* configure dai IPC message */
-	buffer->comp.hdr.size = sizeof(*buffer);
+	buffer->comp.hdr.size = ipc_size;
 	buffer->comp.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_BUFFER_NEW;
 	buffer->comp.id = swidget->comp_id;
 	buffer->comp.type = SOF_COMP_BUFFER;
 	buffer->comp.pipeline_id = index;
 	buffer->comp.core = core;
 
+	/* append extended data to the end of the buffer */
+	memcpy((u8 *)buffer + sizeof(*buffer),
+	       &swidget->comp_ext, sizeof(swidget->comp_ext));
```

Comment:
Is the buffer the proper size for this memcpy to happen already?

Reply:
oops, a copy-paste error, should be this in the allocation:
buffer = kzalloc(ipc_size, GFP_KERNEL);

Reply:
addressed.